creation/web_base/cvmfs_helper_funcs.sh

```diff
@@ -30,476 +30,522 @@
-            # evaluate the worker node's system configurations to decide whether CVMFS can be mounted or not
-            loginfo "Evaluating the worker node..."
-            # display operating system information
-            print_os_info
-
-        # assess the worker node based on its existing system configurations and perform next steps accordingly
-        if evaluate_worker_node_config ; then
-            # if evaluation was true, then proceed to mount CVMFS
-            if [[ $glidein_cvmfs = never ]]; then
-                # do nothing; test the node and print the results but do not even try to mount CVMFS
-                # just continue with glidein startup
-                echo $?
-                "$error_gen" -ok "`basename $0`" "msg" "Not trying to install CVMFS."
+perform_cvmfs_mount () {
+    # by this point, it would have been established that CVMFS is not locally available
+    # so, install CVMFS via mode 1 of cvmfsexec (mountrepo) or mode 3 of cvmfsexec (cvmfsexec)
+    loginfo "CVMFS is NOT locally mounted on the worker node! Mounting now..."
+    # check the operating system distribution
+    #if [[ $GWMS_OS_DISTRO = RHEL ]]; then
+    # evaluate the worker node's system configurations to decide whether CVMFS can be mounted or not
+    loginfo "Evaluating the worker node..."
+    # display operating system information
+    print_os_info
+
+    local cvmfsexec_mode=$1
+    # assess the worker node based on its existing system configurations and perform next steps accordingly
+    if evaluate_worker_node_config ; then
+        # if evaluation was true, then proceed to mount CVMFS
+        if [[ $glidein_cvmfs = never ]]; then
+            # do nothing; test the node and print the results but do not even try to mount CVMFS
+            # just continue with glidein startup
+            echo $?
+            "$error_gen" -ok "$(basename $0)" "msg" "Not trying to install CVMFS."
+        else
+            loginfo "Mounting CVMFS repositories..."
+            if mount_cvmfs_repos $cvmfsexec_mode $GLIDEIN_CVMFS_CONFIG_REPO $GLIDEIN_CVMFS_REPOS ; then
+                :
             else
-                loginfo "Mounting CVMFS repositories..."
-                if mount_cvmfs_repos $GLIDEIN_CVMFS_CONFIG_REPO $GLIDEIN_CVMFS_REPOS ; then
-                    :
+                if [[ $glidein_cvmfs = required ]]; then
+                    # if mount CVMFS is not successful, report an error and exit with failure exit code
+                    echo $?
+                    "$error_gen" -error "$(basename $0)" "WN_Resource" "CVMFS is required but unable to mount CVMFS on the worker node."
+                    exit 1
+                elif [[ $glidein_cvmfs = preferred || $glidein_cvmfs = optional ]]; then
+                    # if mount CVMFS is not successful, report a warning/error in the logs and continue with glidein startup
+                    # script status must be OK, otherwise the glidein will fail
+                    echo $?
```

Comment:
remove echo $? will alwyas return a zero because of the previous if condition always evaluating to zero

if wanting to print to stdout, then use echo 0/1